test/integration/grpc_datastore_test.dart

```diff
@@ -30,14 +30,19 @@ main() async {
       (String project, ServiceAccountCredentials serviceAccount) async {
     final accessTokenProvider =
         ServiceAccountTokenProvider(serviceAccount, scopes);
-    final dialer = grpc.Dialer(endpoint);
-    final client = grpc.Client(dialer, accessTokenProvider, 10);
-    final datastore = GrpcDatastoreImpl(client, project);
+    final clientChannel = grpc.ClientChannel(
+      endpoint,
+      options: grpc.ChannelOptions(
+        // TODO(domesticmouse): Attach accessTokenProvider
+        credentials: grpc.ChannelCredentials.insecure(),
```

Comment:
Yeah, I had to look around a bit to figure this out too, there is some authenticators in `package:grpc`, but it's not obvious how to use them, luckily there is an example:

https://github.com/grpc/grpc-dart/blob/f3e1e32973f6f40471b398625f3d1372f93a70cd/example/googleapis/bin/logging.dart#L39-L45

It's not intuitive to create an authenticator and then call `toCallOptions` on it :)

Probably we'll have to use the [ComputeEngineAuthenticator](https://pub.dev/documentation/grpc/latest/grpc/ComputeEngineAuthenticator-class.html) when running in AppEngine, as this will (hopefully) use the metadata service. When running locally we'll probably have to use the [ServiceAccountAuthenticator](https://pub.dev/documentation/grpc/latest/grpc/ServiceAccountAuthenticator-class.html).

Reply:
I've added in code to call `ServiceAccountAuthenticator` or `ComputeEngineAuthenticator` for the two GRPC based services. Well, I think I have, in as much as all the types seem to line up, but I don't have a testing strategy. 

PTAL and let me know what's next. Thanks!